tests/chunked_worker.py

```diff
@@ -0,0 +1,103 @@
+import collections
+import itertools
+import typing
+import uuid
+
+from dss.events.chunkedtask import Runner, Runtime, Task
+
+
+class TestStingyRuntime(Runtime[dict, typing.Any]):
+    """This is runtime that returns a pre-determined sequence, and then 0s for the remaining time."""
+    def __init__(
+            self,
+            global_results: typing.Mapping[str, typing.Any],
+            seq: typing.Optional[typing.Iterable[int]]=None) -> None:
+        self.global_results = global_results
+        self.result = None  # type: typing.Optional[typing.Any]
+        if seq is None:
+            seq = list()
+        self.seq = itertools.chain(seq, itertools.repeat(0))
+        self.scheduled_work = list()  \
+            # type: typing.List[typing.Tuple[typing.Type[Task], dict, typing.Optional[str]]]
+
+    def get_remaining_time_in_millis(self) -> int:
+        return self.seq.__next__()
+
+    def schedule_work(
+            self,
+            task_class: typing.Type[Task[typing.Any, typing.Any]],
+            state: typing.Any,
+            new_task: bool,
+    ) -> str:
+        # it's illegal for there to be no state.
+        assert state is not None
+
+        task_id = str(uuid.uuid4()) if new_task else None
+        self.scheduled_work.append((task_class, state, task_id))
+        return task_id
+
+    def work_complete_callback(self, result: typing.Any):
+        self.result = result
+
+
+def task_lifecycle(
```

Comment:
Can we name this something more action-y like `advance_task_lifecycle()` or `process_task_updates()`?